File: CSE 210 - Programming with Classes/prove/Develop02/JournalProgram.py
import random
from datetime import datetime

class Journal:
    def __init__(self):
        self.prompts = {
            "How was your day?": False,
            "How is your family?": False,
            "What was the weather like today?": False,
            "What was fun today?": False,
            "How was work today?": False
        }
        self.responses = []
    
    def show_menu(self):
        print("\nMenu")
        print("1. Write a new entry")
        print("2. Display the journal")
        print("3. Exit")
    
    def display_entry(self):
        if not self.responses:
            print("\nNo journal entries yet.")
        else:
            print("\nJournal Entries:")
            for response in self.responses:
                print(response)
    
    def write(self):
        unanswered_questions = [q for q, answered in self.prompts.items() if not answered]
        if not unanswered_questions:
            print("\nNo more prompts available.")
            return
        
        random_question = random.choice(unanswered_questions)
        self.prompts[random_question] = True
        print(f"\n{random_question}")
        
        answer = input("Your response: ").strip()
        while not answer:
            print("Response cannot be empty.")
            answer = input("Your response: ").strip()
        
        date = datetime.now().strftime("%d/%m/%Y")
        self.responses.append(f"Date: {date} - Prompt: {random_question} - {answer}")
    
    def run(self):
        while True:
            self.show_menu()
            choice = input("\nEnter your choice: ").strip()
            
            if choice == "1":
                self.write()
            elif choice == "2":
                self.display_entry()
            elif choice == "3":
                print("\nExiting the journal. Goodbye!")
                break
            else:
                print("\nInvalid choice. Please try again.")

# Run the journal
if __name__ == "__main__":
    journal = Journal()
    journal.run()

File: CSE 210 - Programming with Classes/prove/Develop02/JournalProgram.cs
using System;
using System.Collections.Generic;

public class Journal
{
    private Dictionary<string, bool> prompts;
    private List<string> responses;

    public Journal()
    {
        prompts = new Dictionary<string, bool>
        {
            { "How was your day?", false },
            { "How is your family?", false },
            { "What was the weather like today?", false },
            { "What was fun today?", false },
            { "How was work today?", false }
        };

        responses = new List<string>();
    }

    public void ShowMenu()
    {
        Console.WriteLine("\nMenu");
        Console.WriteLine("1. Write a new entry");
        Console.WriteLine("2. Display the journal");
        Console.WriteLine("3. Exit");
    }

    public void DisplayEntry()
    {
        if (responses.Count == 0)
        {
            Console.WriteLine("\nNo journal entries yet.");
        }
        else
        {
            Console.WriteLine("\nJournal Entries:");
            foreach (var response in responses)
            {
                Console.WriteLine(response);
            }
        }
    }

    public void Write()
    {
        // Find unanswered questions
        var unansweredQuestions = new List<string>();
        foreach (var prompt in prompts)
        {
            if (!prompt.Value)
            {
                unansweredQuestions.Add(prompt.Key);
            }
        }

        if (unansweredQuestions.Count == 0)
        {
            Console.WriteLine("\nNo more prompts available.");
            return;
        }

        Random random = new Random();
        string randomQuestion = unansweredQuestions[random.Next(unansweredQuestions.Count)];
        prompts[randomQuestion] = true; // Mark the prompt as answered
        Console.WriteLine($"\n{randomQuestion}");

        string answer = Console.ReadLine().Trim();

        // Ensure the answer is not empty
        while (string.IsNullOrEmpty(answer))
        {
            Console.WriteLine("Response cannot be empty.");
            answer = Console.ReadLine().Trim();
        }

        string date = DateTime.Now.ToString("dd/MM/yyyy");
        responses.Add($"Date: {date} - Prompt: {randomQuestion} - {answer}");
    }

    public void Run()
    {
        while (true)
        {
            ShowMenu();
            Console.Write("\nEnter your choice: ");
            string choice = Console.ReadLine().Trim();

            if (choice == "1")
            {
                Write();
            }
            else if (choice == "2")
            {
                DisplayEntry();
            }
            else if (choice == "3")
            {
                Console.WriteLine("\nExiting the journal. Goodbye!");
                break;
            }
            else
            {
                Console.WriteLine("\nInvalid choice. Please try again.");
            }
        }
    }

    // Main method to run the journal program
    public static void Main(string[] args)
    {
        Journal journal = new Journal();
        journal.Run();
    }
}
